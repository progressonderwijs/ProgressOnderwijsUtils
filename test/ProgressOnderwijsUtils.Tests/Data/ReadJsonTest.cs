using System.Threading.Tasks;

namespace ProgressOnderwijsUtils.Tests.Data;

public sealed class ReadJsonTest : TransactedLocalConnection
{
    [Fact]
    public async Task ToDo()
    {
        SQL(
            $@"
                create table #ReadJsonTest (
                    ReadJsonTestId int not null
                    , BooleanColumn bit null
                    , NumberColumn int null
                    , StringColumn nvarchar(32) null
                );
            "
        ).ExecuteNonQuery(Connection);

        SQL(
            $@"
                insert into #ReadJsonTest values
                    (1, {true}, 17, 'iets'),
                    (2, null, null, null);
            "
        ).ExecuteNonQuery(Connection);

        using var stream = new MemoryStream();
        await SQL($"select t.* from #ReadJsonTest t").ReadJsonAsync(Connection, stream, CancellationToken.None);

        ApprovalTest.CreateHere().AssertUnchangedAndSave(Encoding.UTF8.GetString(stream.ToArray()));
    }
}
