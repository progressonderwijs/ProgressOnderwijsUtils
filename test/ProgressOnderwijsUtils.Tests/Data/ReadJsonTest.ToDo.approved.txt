[
  {
    "ReadJsonTestId": 1,
    "BooleanColumn": true,
    "NumberColumn": 17,
    "StringColumn": "iets"
  },
  {
    "ReadJsonTestId": 2,
    "BooleanColumn": null,
    "NumberColumn": null,
    "StringColumn": null
  }
]